resolver/resolver.go

```diff
@@ -319,3 +319,12 @@ type Resolver interface {
 	// Close closes the resolver.
 	Close()
 }
+
+// AuthorityOverrider is implemented by Builders that wish to override the
+// default authority for the ClientConn.
+// By default, the authority used is target.Endpoint()
+type AuthorityOverrider interface {
+	// GetServiceAuthority returns the authority to use for a ClientConn with the
+	// given target. It must not perform I/O or any other blocking operations.
+	GetServiceAuthority(t Target) string
```

Comment:
Please name the parameter `target` to be explicit, or don't name it at all, as `t` doesn't really add documentation value here, and it's not necessary to name.